pkg/cluster/pod.go

```diff
@@ -91,20 +113,124 @@ func (c *Cluster) registerPodSubscriber(podName spec.NamespacedName) chan spec.P
 	return ch
 }
 
-func (c *Cluster) recreatePod(pod v1.Pod) error {
+func (c *Cluster) movePod(pod *v1.Pod) (*v1.Pod, error) {
 	podName := util.NameFromMeta(pod.ObjectMeta)
+	if err := c.recreatePod(podName); err != nil {
+		return nil, fmt.Errorf("could not move old master pod: %v", err)
+	}
+
+	newPod, err := c.KubeClient.Pods(podName.Namespace).Get(podName.Name, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("could not get pod: %v", err)
+	}
+
+	if newPod.Spec.NodeName == pod.Spec.NodeName {
+		return nil, fmt.Errorf("pod %q remained on the same node", podName)
+	}
+
+	node, err := c.KubeClient.Nodes().Get(newPod.Spec.NodeName, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("could not get node %q: %v", pod.Spec.NodeName, err)
+	}
+
+	if util.MapContains(node.Labels, c.OpConfig.CordonedNodeLabels) {
```

Comment:
If we set PreferredDuringSchedulingIgnoredDuringExecution then it's ok to land on the CordonedNode, otherwise (if we choose the Required... ) then this check is fine.

Reply:
it's better to log this message and continue in any case.